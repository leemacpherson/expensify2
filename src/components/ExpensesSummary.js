import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpenseTotal = numeral(expenseTotal / 100).format("$0,0.00");
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}.
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const filteredExpenses = selectExpenses(state.expenses, state.filters);
  console.log("filteredExpenses is ", filteredExpenses);
  return {
    expenseCount: filteredExpenses.length,
    expenseTotal: selectExpensesTotal(filteredExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
