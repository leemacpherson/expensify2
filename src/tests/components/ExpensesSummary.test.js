import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should match snapshot for no expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={0} expenseTotal={0} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should match snapshot for 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expenseTotal={1000} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should match snapshot for 3 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={3} expenseTotal={10009} />
  );
  expect(wrapper).toMatchSnapshot();
});
