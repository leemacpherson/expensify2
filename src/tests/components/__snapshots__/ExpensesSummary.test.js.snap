// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $10.00
    .
  </h1>
</div>
`;

exports[`should match snapshot for 3 expense 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $100.09
    .
  </h1>
</div>
`;

exports[`should match snapshot for no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
    .
  </h1>
</div>
`;
